test/test.js

```diff
@@ -458,6 +462,37 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+    pathname === '/browserWorkerTestReports') {
+
+    var body = '';
+
+    req.on('data', function (chunk) {
+      body += chunk;
+    });
+
+    req.on('end', function () {
+      var map = libCoverage.createCoverageMap(JSON.parse(body));
+      var summary = libCoverage.createCoverageSummary();
+      map.merge(summary);
+
+      map.files().forEach(function (f) {
+        var fc = map.fileCoverageFor(f),
+          s = fc.toSummary();
+        summary.merge(s);
+      });
+      fs.writeFileSync(
+        '../coverage/coverageinfo.json', summary);
```

Comment:
I would be surprised if this works, because `summary` is a [`CoverageSummary` object](https://github.com/istanbuljs/istanbuljs/blob/45936de1356b743dcf82df729bf3da895e180aea/packages/istanbul-lib-coverage/lib/file.js#L45-L97), while `writeFileSync` expects a string or `Buffer`.
 You should use `JSON.stringify(summary)` to serialize it before it can be written.